wear/src/main/java/com/example/k_dual/watchface/KDualCanvasRender.kt

```diff
@@ -53,6 +70,20 @@ class KDualCanvasRender (
     private val robotoMedium = Typeface.createFromAsset(assetManager, "Roboto-Medium.ttf")
     private val robotoRegular = Typeface.createFromAsset(assetManager, "Roboto-Regular.ttf")
 
+    // Get watch battery stat
+    private class BatteryReceiver: BroadcastReceiver() {
+        var batteryStat: Int = 0
+        override fun onReceive(p0: Context?, intent: Intent?) {
+            batteryStat = intent?.getIntExtra(BatteryManager.EXTRA_LEVEL, -1) ?: 0
+        }
+    }
+    private val intentFilter = IntentFilter(Intent.ACTION_BATTERY_CHANGED)
+    private val batteryReceiver = BatteryReceiver()
+
+    init {
+        context.registerReceiver(batteryReceiver, intentFilter)
```

Comment:
[ChatGPT] **Unregistering Receiver**: You've registered a `batteryReceiver` in the init block, but you need to make sure to unregister it to prevent memory leaks. This should be done in the `onDestroy` method of the Renderer:
```kotlin
override fun onDestroy() {
    super.onDestroy()
    context.unregisterReceiver(batteryReceiver)
}
```